### PR TITLE
Ignore celery retry

### PR DESCRIPTION
**Pull request by jcummins54**

### CODE REVIEW

File: ddtrace/contrib/celery/task.py [hunk 1]
```diff
@@ -84,6 +84,12 @@ def _task_init(func, task, args, kwargs):
 
 @require_pin
 def _task_run(pin, func, task, args, kwargs):
+    if task.request.hostname is not None:
+        try:
+            global WORKER_SERVICE
+            WORKER_SERVICE = 'celery_{}'.format(task.request.hostname.split('@')[0])
+        except AttributeError:
+            pass
```
Comment (jd) [inline ddtrace/contrib/celery/task.py]: This seems unrelated to the topic of the PR.

I also don't think you want to change the global value of `WORKER_SERVICE` here or you'll end up updating it on a per-app basis rather than per-task.

What's the name of `hostname` usually?

File: ddtrace/contrib/celery/task.py [hunk 2]
```diff
@@ -111,6 +123,11 @@ def _task_apply(pin, func, task, args, kwargs):
 
 @require_pin
 def _task_apply_async(pin, func, task, args, kwargs):
+    try:
+        global PRODUCER_SERVICE
+        PRODUCER_SERVICE = 'celery_{}'.format(task.request.hostname.split('@')[0])
```
Comment (jd) [inline ddtrace/contrib/celery/task.py]: same as above

File: ddtrace/span.py [hunk 3]
```diff
@@ -233,16 +233,18 @@ def set_exc_info(self, exc_type, exc_val, exc_tb):
         if not (exc_type and exc_val and exc_tb):
             return # nothing to do
 
-        self.error = 1
+        # readable version of type (e.g. exceptions.ZeroDivisionError)
+        exc_type_str = "%s.%s" % (exc_type.__module__, exc_type.__name__)
+
+        # do not treat celery.exceptions.Retry as an error, as it is expected behavior
+        if exc_type_str != 'celery.exceptions.Retry':
```
Comment (jd) [inline ddtrace/span.py]: You can't add celery specific stuff in this file. This should be done at a different level.

**Comment by Kyle-Verhoog** [comment]
It looks like we have test coverage that Celery retries are not tagged as errors. 
https://github.com/DataDog/dd-trace-py/blob/d6e7d699b98315470f12b4261a5c61a7adb32aa9/tests/contrib/celery/test_integration.py#L261-L264

Going to close this for now. Although I'm happy to reopen if the problem still persists!
